makefiles/boot/riotboot.mk

```diff
@@ -0,0 +1,110 @@
+ifneq (,$(filter riotboot,$(FEATURES_PROVIDED)))
+
+.PHONY: riotboot/flash riotboot/flash-bootloader riotboot/flash-slot0 riotboot/bootloader/%
+
+RIOTBOOT_DIR = $(RIOTBASE)/bootloaders/riotboot
+RIOTBOOT ?= $(RIOTBOOT_DIR)/bin/$(BOARD)/riotboot.elf
+CFLAGS += -I$(BINDIR)/riotbuild
+
+HEADER_TOOL_DIR = $(RIOTBASE)/dist/tools/riot_hdr
+HEADER_TOOL ?= $(HEADER_TOOL_DIR)/bin/genhdr
+
+# Indicate the reserved space for a header, 256B by default
+# Notice that it must be 256B aligned. This is restricted by
+# the Cortex-M0+/3/4/7 architecture
+RIOTBOOT_HDR_LEN ?= 0x100
+
+# export variables for 'slot_util'
+export RIOTBOOT_LEN
+export RIOTBOOT_SLOT_LEN
+
+# Mandatory APP_VER, set to 0 by default
+APP_VER ?= 0
+
+# For slot generation, one needs only to link
+$(BINDIR)/$(APPLICATION)-%.elf: link
+	$(Q)$(_LINK) -o $@
+
+# slot 0 targets
+SLOT0_OFFSET := $$(($(RIOTBOOT_LEN) + $(RIOTBOOT_HDR_LEN)))
+$(BINDIR)/$(APPLICATION)-slot%.elf: FW_ROM_LEN=$(RIOTBOOT_SLOT_LEN - $(RIOTBOOT_HDR_LEN))
```

Comment:
Same confusion here, this does not generates the `slot` but the `slot.firmware`. 

I would rename them to

`-slot%.firmware.bin` (or `fw.bin`), `-slot%.hdr.bin` and `slot%.bin`.

Reply:
> this does not generates the slot but the slot.firmware

I don't fully understand this. Maybe it's because I messed up with the definitions, but a slot should definitely be composed of header + firmware.